builtin/aws/ecr/pull/builder.go

```diff
@@ -0,0 +1,181 @@
+`)
+
+	doc.Example(`
+build {
+  use "aws-ecr-pull" {
+	region     = "us-east-1"
+    repository = "deno-http"
+    tag        = "latest"
+  }
+}
+`)
+
+	doc.Input("component.Source")
+	doc.Output("ecr.Image")
+
+	doc.SetField(
+		"region",
+		"the AWS region the ECR repository is in",
+		docs.Summary(
+			"if not set uses the environment variable AWS_REGION or AWS_REGION_DEFAULT.",
+		),
+	)
+
+	doc.SetField(
+		"repository",
+		"the AWS ECR repository name",
+	)
+
+	doc.SetField(
+		"tag",
+		"the tag of the image to pull",
+	)
+	return doc, nil
+}
+
+// Config implements Configurable
+func (b *Builder) Config() (interface{}, error) {
+	return &b.config, nil
+}
+
+// Build
+func (b *Builder) Build(ui terminal.UI, log hclog.Logger) (*ecr.Image, error) {
+
+	// If there is no region setup. Try and load it from environment variables.
+	if b.config.Region == "" {
+		b.config.Region = os.Getenv("AWS_REGION")
+
+		if b.config.Region == "" {
+			b.config.Region = os.Getenv("AWS_REGION_DEFAULT")
+		}
+	}
+
+	if b.config.Region == "" {
+		return nil, status.Error(
+			codes.FailedPrecondition,
+			"Please set your aws region in the deployment config, or set the environment variable 'AWS_REGION' or 'AWS_DEFAULT_REGION'")
+	}
+
+	sg := ui.StepGroup()
+	step := sg.Add("")
+	defer func() {
+		if step != nil {
+			step.Abort()
+		}
+	}()
+
+	step.Update("Connecting to AWS")
+	sess, err := utils.GetSession(&utils.SessionConfig{
+		Region: b.config.Region,
+		Logger: log,
+	})
+
+	if err != nil {
+		log.Error("error connecting to AWS", "error", err)
+		return nil, err
+	}
+
+	step.Done()
+
+	step = sg.Add("Verifying image exists")
+	svc := awsecr.New(sess)
+
+	cfgTag := b.config.Tag
+	cfgRepository := b.config.Repository
+
+	imgs, err := svc.DescribeImages(&awsecr.DescribeImagesInput{
+		RepositoryName: aws.String(cfgRepository),
+		Filter: &awsecr.DescribeImagesFilter{
+			TagStatus: aws.String("TAGGED"),
+		},
+	})
+	if err != nil {
+		log.Error("error describing images", "error", err, "repository", cfgRepository)
+		return nil, err
+	}
+
+	if len(imgs.ImageDetails) == 0 {
+		log.Error("no tagged images found", "repository", cfgRepository)
+		return nil, status.Error(codes.FailedPrecondition, "No images found")
+	}
+	log.Debug("found images", "image count", len(imgs.ImageDetails))
+
+	var output ecr.Image
+	for _, img := range imgs.ImageDetails {
+		for _, tag := range img.ImageTags {
+			if *tag == cfgTag {
+
+				output.Image = *img.RegistryId + ".dkr.ecr." + b.config.Region + ".amazonaws.com/" + cfgRepository
+				output.Tag = *tag
+				// TODO(kevinwang): Do we need to get architecture?
+				// If we do, we can pull the image and inspect it via `cli.ImageInspectWithRaw`,
+				// - prior art: /builtin/docker/builder.go -> Build
+				// There is also an open issue for the ECR team to build a architecture feature into
+				// the UI, which probably comes with a CLI/API change.
+				// - see https://github.com/aws/containers-roadmap/issues/1591
```

Comment:
Since ECR doesn't expose the architecture in an obvious way here yet, I'd probably make an option on the plugin called `force_architecture` that will set the architecture on output. This will make it at least work with the other bits we've been adding architecture to.

Reply:
Added `force_architecture` in 20bdea44c1e4618ea014dbb829d919a7d8032953

`force_architecture = "x86_64"`
![CleanShot 2022-06-02 at 00 35 58@2x](https://user-images.githubusercontent.com/26389321/171553796-9bb6baa4-0ef9-4ed3-808c-02204fadee00.png)


`force_architecture` omitted
![CleanShot 2022-06-02 at 00 35 08@2x](https://user-images.githubusercontent.com/26389321/171553730-8f9e04d9-dc1c-4bc3-9fda-218035ffc6c1.png)